### DISCUSSION TITLE
sideloading apk files

### DISCUSSION THREAD

**Discussion by b0nko** [dnschneid/crouton#4690] [Q&A] [answered]
Can I use the xfce version of crouton to run apk files on my chromebook? My chromebook doesn't have built-in Linux support so I was hoping I could install crouton to do this.
Also, is there a visual version of crouton other than unity because when I tried installing unity, it said it wasn't compatible with my ARM chromebook or something

**Comment by TjDaHacker** [comment]
Do you know how to hack i-ready? pls i dont wanna do it

**Reply by DennisLfromGA** [reply]
@TjDaHacker,

That's off-topic, please don't hijack threads.   

-**DennisLfromGA**

**Comment by DennisLfromGA** [comment, marked as answer]
@b0nko,

> Can I use the xfce version of crouton to run apk files on my chromebook? My chromebook doesn't have built-in Linux support so I was hoping I could install crouton to do this.   

If your device supports Google Play then once you're in Developer mode you can sideload apk files, no crouton chroot needed.   

> Also, is there a visual version of crouton other than unity because when I tried installing unity, it said it wasn't compatible with my ARM chromebook or something.

There are several other desktop targets to choose from, just run the following command to see them:  
- `crouton -t list`   

*Hope this helps,*
-**DennisLfromGA**

**Comment by TomTravis** [comment]
yes you can do this
the debootstrap chroots are many, I use debian mate for my work, put the apk package in downloads and install on the chromeOS side

 **testC -r list**
Recognized debian releases:
    potato! woody! sarge! etch! lenny! squeeze! wheezy! jessie! stretch! buster
    **bullseye bookworm** sid
Recognized kali releases:
    moto! kali! sana! **kali-rolling**
Recognized ubuntu releases:
    warty! hoary! breezy! dapper! edgy! feisty! gutsy! hardy! intrepid! jaunty!
    karmic! lucid! maverick! natty! oneiric! precise! quantal! raring! saucy!
    trusty! utopic! vivid! wily! xenial yakkety! zesty! artful! bionic* cosmic!
    disco! eoan! focal* groovy! hirsute! Impish* **jammy**
Releases marked with ! are upstream end-of-life, and should be avoided.
Releases marked with * are unsupported, but may work with some effort.
chronos@localhost ~ $ testC -t list
Available targets:
**Python**
        Basic CLI tools such as ssh.
        Requires: core
audio
        Support audio playback via Chromium OS's audio system.
        Requires: core
chrome
        Google Chrome browser, stable channel.
        Requires: x11
chrome-beta
        Google Chrome browser, beta channel.
        Requires: x11
chrome-dev
        Google Chrome browser, dev channel.
        Requires: x11
chromium
        Chromium browser. Uses the distro's version, which may be old.
        Requires: x11
cli-extra
        Basic CLI tools such as ssh.
        Requires: core
core
        Performs core system configuration. Most users would want this.
e17
        Installs the enlightenment desktop environment. (Approx. 50MB)
        Requires: gtk-extra
extension
        Clipboard synchronization and URL handling with Chromium OS.
        Requires: x11
gnome
        Installs the GNOME desktop environment. (Approx. 400MB)
        Requires: gtk-extra
gnome-desktop
        Installs GNOME along with common applications. (Approx. 1100MB)
        Requires: gnome
gtk-extra
        GTK-based tools including gdebi and a simple browser.
        Requires: x11
kde
        Installs a minimal KDE desktop environment. (Approx. 600MB)
        Requires: x11
kde-desktop
        Installs KDE along with common applications. (Approx. 1000MB)
        Requires: kde
keyboard
        Adds support for Chromebook keyboard special keys.
        Requires: x11
kodi
        Installs the KODI media player. (Approx. 200MB)
        Requires: x11
**libreoffice**
        Libreoffice.
        Requires: core
lxde
        Installs the LXDE desktop environment. (Approx. 200MB)
        Requires: gtk-extra
lxde-desktop
        Installs LXDE along with common applications. (Approx. 800MB)
        Requires: lxde
**mate**
        Installs the mate desktop environment. (Approx. 250MB)
        Requires: gtk-extra
**mate-desktop**
        Installs mate along with common applications. (Approx. 1200MB)
        Requires: mate
**tools**
        tools.
        Requires: core
touch
        Touchscreen and limited generic gesture support.
        Requires: x11
unity
        Installs the Unity desktop environment. (Approx. 700MB)
        Requires: gtk-extra
unity-desktop
        Installs Unity along with common applications. (Approx. 1100MB)
        Requires: unity
x11
        X11 via autodetected backend. Does not install any desktop environment.
xfce
        Installs the Xfce desktop environment. (Approx. 250MB)
        Requires: gtk-extra
xfce-desktop
        Installs Xfce along with common applications. (Approx. 1200MB)
        Requires: xfce
xiwi
        X.org X11 backend running unaccelerated in a Chromium OS window.
        Requires: audio extension
xorg
        X.Org X11 backend. Enables GPU acceleration on supported platforms.
        Requires: audio
be aware using shell to install may not be the best idea for tools and Desktops

my shell update
[testC](https://drive.google.com/file/d/1g1MEwx0wFBX8HAUsaHGNfnD6uXiGageq/view?usp=sharing) and download to install

update the apt /etc/apt sources.list file and use sudo apt-get install or git repo and build for python rust ninja meson go this is what I do
